### PR TITLE
[TF FE] SparseSegmentMean translator

### PR DESCRIPTION
**Pull request by popovaan**
### Details:
 - Added translator for SparseSegmentMean 

### Tickets:
 - 149705

### CODE REVIEW

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        [5, 3],
+    ])
+    @pytest.mark.precommit
+    @pytest.mark.nightly
+    def test_sparse_segment_mean(self, data_type, indices_type,
+                                shape, segments_num,
+                                ie_device, precision, ir_version, temp_dir,
+                                use_legacy_frontend):
+        if ie_device == 'GPU':
+            pytest.skip("GPU error: to_shape was called on a dynamic shape")
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:72]: please create bug in JIRA for them and indicate ticket number here.

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            tf.raw_ops.SparseSegmentMean(
+                data=values,
+                indices=indices,
+                segment_ids=segments_ids)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        ref_net = None
+
+        return tf_net, ref_net
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:57]:
```suggestion
        return tf_net, None
```

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py (hunk 1 above)
Comment (popovaan) [reply tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py]: Done.

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                data=values,
+                indices=indices,
+                segment_ids=segments_ids)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    @pytest.mark.parametrize('data_type', [np.float32, np.float64, np.float16])
+    @pytest.mark.parametrize('indices_type', [np.int32, np.int64])
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:58]: please have parameter type for `indices` and `segment_ids` separately. They may have different types, for example. one - of int32 and other - of int64

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+            tf.raw_ops.SparseSegmentMean(
+                data=values,
+                indices=indices,
+                segment_ids=segments_ids)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    @pytest.mark.parametrize('data_type', [np.float32, np.float64, np.float16])
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:57]:
```suggestion
    @pytest.mark.parametrize('data_type', [np.float16, np.float32, np.float64])
```

File: src/frontends/tensorflow_common/src/op/sparse_segment_mean.cpp [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+namespace tensorflow {
+namespace op {
+OutputVector translate_sparse_segment_mean_op(const NodeContext& node) {
+    default_op_checks(node, 3, {"SparseSegmentMean"});
+    auto data = node.get_input(0);
+    auto indices = node.get_input(1);
+    auto segment_ids = node.get_input(2);
+
+    // get the last index from segment_ids
+    auto segments_ids_size = std::make_shared<v3::ShapeOf>(segment_ids, indices.get_element_type());
```
Comment (rkazants) [inline src/frontends/tensorflow_common/src/op/sparse_segment_mean.cpp:33]: `indices.get_element_type()` can be undefined at this stage. Please use concrete time `i64`

Comment (rkazants) [reply src/frontends/tensorflow_common/src/op/sparse_segment_mean.cpp:33]: propose to convert both `indices` and `segment_ids` to `i64` and use this fact below so no need to call `get_element_type()`

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
+# Copyright (C) 2018-2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import random
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:3]: please use rng for integer generation

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+                indices=indices,
+                segment_ids=segments_ids)
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    @pytest.mark.parametrize('data_type', [np.float32, np.float64, np.float16])
+    @pytest.mark.parametrize('indices_type', [np.int32, np.int64])
+    @pytest.mark.parametrize('shape, segments_num', [
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:59]: please use separate shape for `data` and for `indices`. `data` can be of ND rank

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py (hunk 3 above)
Comment (popovaan) [reply tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:59]: Done.

File: src/frontends/tensorflow_common/src/op/sparse_segment_mean.cpp (hunk 5 above)
Comment (popovaan) [reply src/frontends/tensorflow_common/src/op/sparse_segment_mean.cpp]: Done.

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py (hunk 6 above)
Comment (popovaan) [reply tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py]: Done.

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py (hunk 7 above)
Comment (popovaan) [reply tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py]: Done.

File: tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+            tf.compat.v1.global_variables_initializer()
+            tf_net = sess.graph_def
+
+        return tf_net, None
+
+    @pytest.mark.parametrize('data_type', [np.float16, np.float32, np.float64])
+    @pytest.mark.parametrize('indices_type', [np.int32, np.int64])
+    @pytest.mark.parametrize('segment_indices_type', [np.int32, np.int64])
+    @pytest.mark.parametrize('shape, indices_shape, segments_num', [
+        [10, 7, 8],
```
Comment (rkazants) [inline tests/layer_tests/tensorflow_tests/test_tf_SparseSegmentMean.py:62]: please have shape in a form: [10], [7]. Then you don't need square bracket during creation of placeholder.
Also, please test a case with ND data like:
[[10, 20], [7], 8],
[[10, 2, 4], [10], 4]

**Review by rkazants — APPROVED** [review]
left last comment. Other part looks good.

**Pull request merged**